src/main.rs

```diff
@@ -294,85 +297,95 @@ fn main() {
-        update_cache(&cache, args.quiet, enable_styles);
-        true
-    } else {
-        false
+    // Collect languages
+    let languages = match args.language {
+        Some(ref lang) => vec![lang.to_string()],
+        None => get_languages_from_env(),
     };
 
-    // Check cache presence and freshness
-    if !cache_updated
-        && (args.list || !args.command.is_empty())
-        && check_cache(&cache, &args, enable_styles) == CheckCacheResult::CacheMissing
-    {
-        process::exit(1);
-    }
+    for platform in platforms {
+        // Instantiate cache. This will not yet create the cache directory!
+        let cache = Cache::new(platform, &config.directories.cache_dir.path);
 
-    // List cached commands and exit
-    if args.list {
-        let custom_pages_dir = config
-            .directories
-            .custom_pages_dir
-            .as_ref()
-            .map(PathWithSource::path);
-        println!("{}", cache.list_pages(custom_pages_dir).join("\n"));
-        process::exit(0);
-    }
+        // Clear cache, pass through
+        if args.clear_cache {
+            clear_cache(&cache, args.quiet, enable_styles);
+        }
 
-    // Show command from cache
-    if !args.command.is_empty() {
-        // Note: According to the TLDR client spec, page names must be transparently
-        // lowercased before lookup:
-        // https://github.com/tldr-pages/tldr/blob/main/CLIENT-SPECIFICATION.md#page-names
-        let command = args.command.join("-").to_lowercase();
-
-        // Collect languages
-        let languages = args
-            .language
-            .map_or_else(get_languages_from_env, |lang| vec![lang]);
-
-        // Search for command in cache
-        if let Some(lookup_result) = cache.find_page(
-            &command,
-            &languages,
-            config
+        // Cache update, pass through
+        let cache_updated = if should_update_cache(&cache, &args, &config) {
+            update_cache(&cache, args.quiet, enable_styles);
+            true
+        } else {
+            false
+        };
+
+        // Check cache presence and freshness
+        if !cache_updated
+            && (args.list || !args.command.is_empty())
+            && check_cache(&cache, &args, enable_styles) == CheckCacheResult::CacheMissing
+        {
+            process::exit(1);
+        }
+
+        // List cached commands and exit
+        if args.list {
+            let custom_pages_dir = config
                 .directories
                 .custom_pages_dir
                 .as_ref()
-                .map(PathWithSource::path),
-        ) {
-            if let Err(ref e) =
-                print_page(&lookup_result, args.raw, enable_styles, args.pager, &config)
-            {
-                print_error(enable_styles, e);
-                process::exit(1);
-            }
+                .map(PathWithSource::path);
+            println!("{}", cache.list_pages(custom_pages_dir).join("\n"));
```

Comment:
I wonder if we want to display some disambiguation here: What should the output be, if we run `tldr --platform linux --platform windows --list cd`? @dbrgn 

We should check the client specification from tldr-pages/tldr for that (or ask them for a recommendation)